### PR TITLE
ARGO-137 Multitenant support for ngiAvailability

### PR DESCRIPTION
**Pull request by dpavlos**
- Added token based tenant authentication.
- Modified structs and added Job field. Removed old filtering stuff because now we have the job information which includes them. 
- Field names now are **not** tokenized.
- Updated mongo aggregations accordingly.
- Changed package name and api call to a more generic one.
- Removed `group_type` api parameter.
- Removed deprecated code.
- Added godoc.

Examples:

Old API call and XML response for monthly results:
```
/group_availability?start_time=2015-06-01T00:00:00Z&end_time=2015-06-03T00:00:00Z&availability_profile=test-ap1&group_type=ngi&granularity=monthly
```
```
<Profile name="test-ap1">
    <Ngi NGI="TestA">
         <Availability timestamp="2015-06" availability="99.97347491468719" reliability="99.97347491468719"/>
    </Ngi>
</Profile>
```

New API call and XML response for monthly results:
```
/group_groups_availability?start_time=2015-06-21T00:00:00Z&end_time=2015-06-24T00:00:00Z&job=Job_A&granularity=monthly
```
```
   <Job name="Job_A">
     <SuperGroup name="GROUP_A">
       <Availability timestamp="2015-06" availability="71.75110088070457" reliability="65.61389111289031"></Availability>
     </SuperGroup>
   </Job>
```

Old API Call and XML response for daily results: 
```
/group_availability?start_time=2015-06-01T00:00:00Z&end_time=2015-06-03T00:00:00Z&availability_profile=test-ap1&group_type=ngi&granularity=daily
```
```
<Profile name="test-ap1">
    <Ngi NGI="TestA">
        <Availability timestamp="2015-06-01" availability="99.96700866175979" reliability="99.96700866175979"/>
        <Availability timestamp="2015-06-02" availability="99.9817512713282" reliability="99.9817512713282"/>
    </Ngi>
</Profile>
```

New API call and XML response for daily results: 
```
/group_groups_availability?start_time=2015-06-21T00:00:00Z&end_time=2015-06-24T00:00:00Z&job=Job_A&granularity=daily
```
```
   <Job name="Job_A">
     <SuperGroup name="GROUP_A">
       <Availability timestamp="2015-06-22" availability="68.13896116893515" reliability="50.413931144915935"></Availability>
       <Availability timestamp="2015-06-23" availability="75.36324059247399" reliability="80.8138510808647"></Availability>
     </SuperGroup>
   </Job>
```

### CODE REVIEW

File: app/ngiAvailability/ngiAvailabilityView.go [hunk 1]
```diff
@@ -34,39 +34,39 @@ import (
...
+func createView(results []ApiSuperGroupAvailabilityInProfileOutput, format string) ([]byte, error) {
 
 	docRoot := &Root{}
 
-	prevProfile := ""
-	prevNgi := ""
-	ngi := &Ngi{}
-	profile := &Profile{}
+	prevJob := ""
+	prevSubGroup := ""
```
Comment (pkoro) [inline app/ngiAvailability/ngiAvailabilityView.go]: sed SubGroup with SuperGroup here and below.

Comment (dpavlos) [inline app/ngiAvailability/ngiAvailabilityView.go]: Fixed.

**Comment by dpavlos** [comment]
please review /cc @pkoro @makistsantekidis @kaggis

**Comment by pkoro** [comment]
+1

**Pull request merged**